interp/impargs.ml

```diff
@@ -698,28 +701,80 @@ let declare_manual_implicits local ref ?enriching l =
   let enriching = Option.default flags.auto enriching in
   let autoimpls = compute_auto_implicits env sigma flags enriching t in
   let l' = match l with
-    | [] -> assert false
     | [l] ->
        [DefaultImpArgs, set_manual_implicits flags enriching autoimpls l]
+    | _ -> assert false in
+  let req =
+    if is_local local ref then ImplLocal
+    else ImplInteractive(ref,flags,ImplManual (List.length autoimpls))
+  in add_anonymous_leaf (inImplicits (req,[ref,l']))
+
+let maybe_declare_manual_implicits local ref ?enriching l =
+  match l with
+  | [] -> ()
+  | _ -> declare_manual_implicits local ref ?enriching [l]
+
+(* TODO: either turn these warnings on and document them, or handle these cases sensibly *)
```

Comment:
This needs to be resolved before the PR can be merged (let's try to avoid introducing new undocumented warnings, even disabled by default).